src/cmake_generator.cpp

```diff
@@ -946,6 +960,49 @@ void generate_cmake(const char *path, const parser::Project *parent_project) {
                     auto source_key = condition.empty() ? "sources" : (condition + ".sources");
                     throw std::runtime_error(target.name + " " + source_key + " wildcard found 0 files");
                 }
+
+                // For non-interface targets, ensure there is a source file linked for the project's languages.
+                if (target.type != parser::target_interface) {
+                    // All acceptable extensions based off our given languages.
+                    tsl::ordered_set<std::string> proj_extensions = [&]() {
+                        tsl::ordered_set<std::string> exts;
+                        for (auto it = known_languages.begin(); it != known_languages.end(); ++it) {
+                            if (std::find(project.project_languages.begin(), project.project_languages.end(), it->first) ==
+                                project.project_languages.end()) {
+                                continue;
+                            }
+                            // Add all extensions of this language into the list
+                            for (auto const &ext : it->second) {
+                                exts.insert(ext);
+                            }
+                        }
+                        return exts;
+                    }();
+
+                    bool has_hit_def = false;
+                    for (auto &source : sources) {
+                        fs::path path = source;
+                        if (!path.has_extension()) {
+                            continue;
+                        }
+
+                        auto ext = path.extension().string();
+
+                        // Only test lower-case variant of the extension
+                        std::transform(ext.begin(), ext.end(), ext.begin(), std::tolower);
```

Comment:
```cpp
char asciitolower(char in) {
    if (in <= 'Z' && in >= 'A')
        return in - ('Z' - 'z');
    return in;
}

std::transform(data.begin(), data.end(), data.begin(), asciitolower);
```

Probably should be something like this. Although I think there might be conflicts? Earlier you declared that `.C` files are part of the `CXX` language...